packages/shared-lib/src/app/utils/HttpError.ts

```diff
@@ -31,17 +31,17 @@ export class NotFoundError extends HttpError {
  * @param field - optional
  */
 export class RequiredIdNotFoundError extends HttpError {
-  constructor(field: string = 'id') {
-    super(`Required ${field} is missing`, 400, 'EG-102');
+  constructor(field: string = 'id', message?: string) {
+    super(message ? `Required ${field} is missing: ${message}` : `Required ${field} is missing`, 400, 'EG-102');
   }
 }
 
 /**
  * Invalid body parameters request
  */
 export class InvalidRequestError extends HttpError {
-  constructor() {
-    super('Invalid request', 400, 'EG-102');
+  constructor(message?: string) {
+    super(message ? `Invalid request: ${message}` : 'Invalid request', 400, 'EG-102');
```

Comment:
Maybe the super could take the optional param of perhaps `details` and if present it is appended to the end of the message, that would save a lot of duplicate-ish code in each error type